src/components/templates/PartyMap/components/RoomModalOngoingEvent/RoomModalOngoingEvent.tsx

```diff
@@ -30,6 +30,19 @@ export const RoomModalOngoingEvent: React.FC<RoomModalOngoingEventProps> = ({
     roomEvents.length > 0 &&
     (currentEvent ? currentEvent : roomEvents[0]);
 
+  const joinRoom = useCallback(() => {
+    enterRoom();
+    window.open(getRoomUrl(room.url), "_blank", "noopener,noreferrer");
+  }, [enterRoom, room.url]);
+
+  const triggerAttendance = useCallback(() => {
+    dispatch(retainAttendance(true));
+  }, [dispatch]);
+
+  const clearAttendance = useCallback(() => {
+    dispatch(retainAttendance(false));
+  }, [dispatch]);
+
```

Comment:
These are the same as in `src/components/templates/Camp/components/RoomModalOngoingEvent.tsx`, should we extract them to DRY up the code?

I see that the other place is in the legacy Camp Map component, so it may not be worth doing since we are going to remove it soon. So probably just the same comment as above RE: using the helper from `utils/url` that we have instead of `window.open`

Reply:
We are removing the legacy Camp component asap, so won't worry about DRYing this up for this review, but there's bound to be DRYups in future that we'll access once this is done.